src/main/java/it/gov/pagopa/payhub/auth/repository/ClientRepositoryExtImpl.java

```diff
@@ -0,0 +1,28 @@
+package it.gov.pagopa.payhub.auth.repository;
+
+import it.gov.pagopa.payhub.auth.model.Client;
+import org.springframework.data.mongodb.core.FindAndModifyOptions;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.mongodb.core.query.Update;
+
+public class ClientRepositoryExtImpl implements  ClientRepositoryExt {
+	private final MongoTemplate mongoTemplate;
+
+	public ClientRepositoryExtImpl(MongoTemplate mongoTemplate) { this.mongoTemplate = mongoTemplate;	}
+
+	@Override
+	public Client registerClient(String clientId, String organizationIpaCode, String clientSecret) {
```

Comment:
why don't you just use the repository `save` method?
This way you don't even need for a ClientRepositoryExt class

Reply:
ok, keep it simple...